### PR TITLE
Check formatting in CI

### LINKED ISSUE #10: Add `rustfmt_test` and `rustfmt` test coverage checker
`rustfmt_test` (https://bazelbuild.github.io/rules_rust/rust_fmt.html#rustfmt_test) is a Bazel test rule that performs `rustfmt --check` on a set of targets.

We should add one to Client Rust, and integrate it into our CI `build` job, to be run alongside `checkstyle_test`.

We should also add a test coverage check tool to the `dependencies` repo. Similarly to `checkstyle:test-coverage`, it will verify that every Rust source is covered by a `rustfmt_test`.

### PR DESCRIPTION
**Pull request by dmitrii-ubskii**
## What is the goal of this PR?

We add a check to the Factory CI configuration that ensures that all Rust code has been formatted properly.

## What are the changes implemented in this PR?

We add rustfmt configuration, reformat the project, and ensure that rustfmt had been run in CI.

Closes #10.

### CODE REVIEW

File: BUILD [hunk 1]
```diff
@@ -48,6 +48,13 @@ rust_library(
     ],
 )
 
+filegroup(
+    name = "rustfmt_config",
+    srcs = ["rustfmt.toml"],
+)
```
Comment (dmitrii-ubskii) [inline BUILD:54]: `rustfmt.toml` needs to be visible to the rules_rust bazel aspect.

File: .factory/automation.yml [hunk 2]
```diff
@@ -41,6 +41,7 @@ build:
         bazel build //...
         bazel run @vaticle_dependencies//tool/checkstyle:test-coverage
         bazel test $(bazel query 'kind(checkstyle_test, //...)') --test_output=errors
+        bazel test $(bazel query 'kind(rustfmt_test, //...)') --@rules_rust//:rustfmt.toml=//:rustfmt_config
```
Comment (dmitrii-ubskii) [inline .factory/automation.yml:44]: The only way to provide our custom `rustfmt.toml` to the `rustfmt_test` rules that I found is through the aspect.

File: src/answer/numeric.rs [hunk 3]
```diff
@@ -31,11 +31,19 @@ pub enum Numeric {
 
 impl Numeric {
     pub fn into_i64(self) -> i64 {
-        if let Self::Long(value) = self { value } else { panic!() }
+        if let Self::Long(value) = self {
```
Comment (alexjpwalker) [inline src/answer/numeric.rs:34]: This expansion is definitely uglier than the original - can we configure `rustfmt` to use the original?

File: tests/queries.rs [hunk 4]
```diff
@@ -75,25 +109,37 @@ fn run_insert_query(tx: &mut Transaction, query: &str) {
...
-        .iter().fold(String::new(), |acc, db| acc + db.name.as_str() + ",")
+    println!(
+        "{}",
+        client
+            .databases
+            .all()
+            .await
+            .unwrap_or_else(|err| panic!("An error occurred listing databases: {}", err))
+            .iter()
+            .fold(String::new(), |acc, db| acc + db.name.as_str() + ",")
```
Comment (alexjpwalker) [inline tests/queries.rs:131]: Putting every chained call on a new line is definitely ugly in cases like these - can we tell `rustfmt` to not do this?

**Review by alexjpwalker — CHANGES_REQUESTED** [review]

File: src/answer/numeric.rs (hunk 3 above)
Comment (dmitrii-ubskii) [reply src/answer/numeric.rs:34]: Technically yes -- this is what `use_small_heuristics = "Max"` does -- but this feature is not available in stable yet.

File: tests/queries.rs (hunk 4 above)
Comment (dmitrii-ubskii) [reply tests/queries.rs:131]: That does not appear to be an option.

File: .factory/automation.yml (hunk 2 above)
Comment (dmitrii-ubskii) [reply .factory/automation.yml:44]: The way I understand it, this invocation sets the `rustfmt.toml` `label_flag` provided for `rustfmt_aspect` that is used by `rustfmt_test`.

https://bazel.build/extending/config#label-typed-build-settings
https://github.com/bazelbuild/examples/tree/HEAD/configurations/label_typed_build_setting

**Review by alexjpwalker — APPROVED** [review]

**Pull request merged**